Topic4/src/classes/CashStrategy.java

```diff
@@ -0,0 +1,27 @@
+package classes;
+
+import interfaces.Item;
+import interfaces.PaymentStrategy;
+import java.util.List;
+
+public class CashStrategy implements PaymentStrategy {
+
+	public double pay(List<Item> items) {
+		double total = 0;
+		double value;
+		Item mostExpensiveItem = null;// this product will have a discount of
+										// 90%
+		for (Item item : items) {
+			value = item.getValue();
+			total += value;
+			if (mostExpensiveItem == null || value > mostExpensiveItem.getValue()) {
+				mostExpensiveItem = item;
+			}
+		}
+		System.out.println("Cost of the buy: $" + total + ".");

```

Comment:
The system.out.println are not necessary in a service, in that case you could use a logger like log4j or similar to add debug, information and error messages in a log